py/kubeflow/kubeflow/ci/kfctl_e2e_workflow.py

```diff
@@ -494,12 +494,12 @@ def build(self):
     # Checkout
 
     # create the checkout step
+    # always include the default repo
+    repos = [DEFAULT_REPO + "@HEAD"]
     main_repo = argo_build_util.get_repo_from_prow_env()
-    if not main_repo:
-      logging.info("Prow environment variables for repo not set")
-      main_repo = MAIN_REPO + "@HEAD"
-    logging.info("Main repository: %s", main_repo)
-    repos = [main_repo]
+    if (main_repo and not main_repo.startswith(DEFAULT_REPO)):
```

Comment:
Per discussion I need we need to update this so that we check out the proper version in presubmits